src/bin/bench-streamer.rs

```diff
@@ -54,28 +62,83 @@ fn sink(
     })
 }
 
+macro_rules! socketaddr {
+    ($ip:expr, $port:expr) => {
+        SocketAddr::from((Ipv4Addr::from($ip), $port))
+    };
+    ($str:expr) => {{
+        let a: SocketAddr = $str.parse().unwrap();
+        a
+    }};
+}
+
 fn main() -> Result<()> {
-    let read = UdpSocket::bind("127.0.0.1:0")?;
-    read.set_read_timeout(Some(Duration::new(1, 0)))?;
+    let mut num_sockets = 1usize;
+
+    let matches = App::new("solana-bench-streamer")
+        .arg(
+            Arg::with_name("num-recv-sockets")
+                .short("N")
+                .long("num-recv-sockets")
+                .value_name("NUM")
+                .takes_value(true)
+                .help("Use NUM receive sockets"),
+        )
+        .get_matches();
+
+    if let Some(n) = matches.value_of("num-recv-sockets") {
+        num_sockets = n.to_string().parse().expect("integer");
+    }
+
+    fn bind_to(port: u16) -> UdpSocket {
+        let sock = Socket::new(Domain::ipv4(), Type::dgram(), None).unwrap();
+        let sock_fd = sock.as_raw_fd();
+        setsockopt(sock_fd, ReusePort, &true).unwrap();
+        let addr = socketaddr!(0, port);
+        match sock.bind(&SockAddr::from(addr)) {
+            Ok(_) => sock.into_udp_socket(),
+            Err(err) => {
+                panic!("Failed to bind to {:?}, err: {}", addr, err);
+            }
+        }
+    };
+
+    let mut port = 0;
+    let mut addr = SocketAddr::new(IpAddr::V4(Ipv4Addr::new(0, 0, 0, 0)), 0);
 
-    let addr = read.local_addr()?;
     let exit = Arc::new(AtomicBool::new(false));
     let pack_recycler = PacketRecycler::default();
 
-    let (s_reader, r_reader) = channel();
-    let t_reader = receiver(
-        Arc::new(read),
-        exit.clone(),
-        pack_recycler.clone(),
-        s_reader,
-    );
+    let mut read_channels = Vec::new();
+    let read_threads: Vec<JoinHandle<()>> = (0..num_sockets)
+        .into_iter()
+        .map(|_| {
```

Comment:
Can you change this to a standard for loop (or a `fold()`). Modifying mutable variables from inside a `map()` is frowned upon by snobby functional programmers. You generally shouldn't leverage the coincidence that it's implementation executes it from from left-to-right. It's hard to read and prevents the use of drop-in parallelization like Rayon's `par_iter().map()`.